packages/amplify-cli/src/lib/pull-backend.js

```diff
@@ -0,0 +1,38 @@
+const { initializeEnv } = require('./initialize-env');
+const { constructInputParams, postPullCodeGenCheck } = require('./amplify-service-helper');
+
+async function pullBackend(context) {
+  context.exeInfo = context.amplify.getProjectDetails();
+  context.exeInfo.inputParams = constructInputParams(context);
+  context.print.info('');
+  context.print.info('Pre pull check:');
+  const hasChanges = await context.amplify.showResourceTable();
+  context.print.info('');
+
+  context.exeInfo.forcePush = false;
+  context.exeInfo.restoreBackend = !context.exeInfo.inputParams.amplify.noOverride;
+
+  if (hasChanges && context.exeInfo.restoreBackend) {
+    context.print.warning('Local changes detected.');
+    context.print.warning('Amplify pull will override your local changes with what is currently in the cloud.');
```

Comment:
Pulling changes from the cloud will override your local changes. Are you sure you would like to continue? (Y/n)
If user picks yes:
✔ Successfully pulled backend environment '$envname' from the cloud.

If user picks no:
 'amplify push' eill deploy your local updates. However, this will override upstream changes to this backend environment with your local changes. To merge local and upstream changes, commit all backend code changes to Git, perform a merge, resolve conflicts, and then run 'amplify push'.

Reply:
changed